### ISSUE TITLE
Any possibility to make this "easier"?

### ISSUE THREAD

**Issue by docwilmot** [backdrop-contrib/commerce#19]
I don't have any experience with Commerce or Ubercart for that matter so speaking as a complete newbie, and just opinions.

Most of the comments I've read about comparison between the two have focused on the learning curve for Commerce. So I just installed it for the first time and was completely shocked by how sparse this is out of the box. Is there any possibility, since we're now starting from new, to adapt some aspects of Ubercart or even "Commerce Quickstart" so that this is a bit more "complete" when installed? For example there does not seem to be an actual store, there are no product images as default, there's no rendered product view, no attributes or product variations without going deep in the UI, etcetera. again as a newbie those were things I was expecting.

I then went into the Commerce documents site to figure out how those things that I thought were missing could be added and wow this is complex! I see and understand this is for a different audience, but I wonder how other e-commerce offerings from for example Wordpress handle these expectations out of the box.

FWIW I don't quite like the concept of the quick start install profile approach but is that likely to be ported too?

**Comment by argiepiano** [comment]
Yes, commerce is a bit of a beast to learn. The trickiest thing is figuring out how to display products and Add to Cart buttons, since products are not nodes anymore. In commerce, there are several referenced layers: Products -> Line Item types -> Product displays (node bundle). But there are extensive tutorials online.

When I first learned it, I found that the module Commerce Backoffice was really helpful, as it creates several shortcuts and templates. This should be ported.

**Comment by elisseck** [comment]
Agreed, it's quite complex!

I don't see us porting an "install profile" per-say, but I do think this could be a good candidate for a Backdrop "recipe" module for a generic store setup.

I also think a couple things still need to get ported so we're not quite there yet (hence no tagged release). One thing we're missing for a real store is a port of an actual payment method module. For the current use-case we've ported https://www.drupal.org/project/commerce_stripe_pi as custom from a d7 site without the intention of releasing it, because it's really not in very good shape and we don't want to give folks false confidence in it.

**Comment by docwilmot** [comment]
What I'm asking though is are you planning to keep this port faithful to the original D7 code (which would be understandably easier for upgrades from D7, easier for maintenance) or would you consider to add, out of the box, features unique to Backdrop that make Backdrop Commerce friendlier for the user. Like for example, a store layout, product image fields out of the box, built-in rendered product Views, even merge in commerce Backoffice commerce with Dashboard blocks, etc, and so on. Just spitballing, but hope you understand my query.

I think Backdrop's generally tended to lean towards making things "easier" for the end-user than D7, although again I would totally understand the tradeoff with increased work for the maintainers might not be worth it for such a complex project.

**Comment by elisseck** [comment]
Good question, I think it's going to be a bit of a moving target!

In terms of work we're planning to do right now, our plan is to fix bugs and keep this port _reasonably_ faithful to the D7 code to facilitate migrations from Drupal. As time goes on (maybe the next 1-2 years timescale) and fewer folks are actively migrating from D7 that priority may change.

Largely I think anything is up for discussion if someone is motivated enough to put together a PR... but I would encourage experimenting with that sort of Backdrop "quality of life" features in another contributed module that can sit alongside Commerce and potentially be merged in at a later time. If you end up needing extra hooks or small modifications to Commerce to facilitate that sort of thing I think we'd be happy to merge that sort of PR as well.

**Comment by docwilmot** [comment]
Good answers

**Comment by albanycomputers** [comment]
Just a heads up... I've ported commerce_backoffice and views_megarows - both seem to be working... but needs a bit of work before moving to backdrop_contrib...

Look on my github profile to see my versions.

**Comment by oadaeh** [comment]
Regarding @docwilmot's question and @elisseck's answer, I think it makes the most sense to have any added Backdrop specific functionality be in a sub-module, rather than merging it into the main body of code. With it merged into main body of code, people moving from D7 might have their sites (with any potential customizations) broken, due to the added Backdrop specific functionality. With it in a separate sub-module, that possibility is eliminated, but people starting with Backdrop could easily just enable one (presumably) additional module to get the Backdrop specific functionality, which could be done when installing Commerce.

**Comment by elisseck** [comment]
We've tagged a release for commerce core, so this is moving along. I think it might be a good idea to look into some sort of recipe module that installs demo or commonly used config if we wanted to make things "easier" for folks, which would be along the lines of sub-modules noted above. Please feel free to open another issue with specificity or PR such a module!
